geth/jail/jail_cell.go

```diff
@@ -75,11 +64,39 @@ func (cell *JailCell) Fetch(url string, callback func(otto.Value)) (otto.Value,
 			});
 		});
 	`)
+
+	if err != nil {
+		cell.Unlock()
+		return val, err
+	}
+
+	cell.Unlock()
```

Comment:
Let's move all that internal logic but the last return to a private function. So that `defer cell.Unlock()` is reliably called inside it.